dask_cuda/proxify_host_file.py

```diff
@@ -123,31 +152,45 @@ class ProxiesOnDevice(Proxies):
     handle that multiple proxy objects can refer to the same underlying
     device memory object. Thus, we have to track aliasing and make sure
     we don't count down the memory usage prematurely.
+
+    Notice, we only track direct aliasing thus multiple proxy objects can
+    point to different non-overlapping parts of the same device buffer.
+    In this case the tally of the total device memory usage is incorrect.
     """
 
     def __init__(self):
         super().__init__()
-        self.proxy_id_to_dev_mems: Dict[int, Set[Hashable]] = {}
-        self.dev_mem_to_proxy_ids: DefaultDict[Hashable, Set[int]] = defaultdict(set)
+        self.proxy_id_to_dev_mems: Dict[int, Set[DeviceMemoryId]] = {}
+        self.dev_mem_to_proxy_ids: DefaultDict[DeviceMemoryId, Set[int]] = defaultdict(
+            set
+        )
 
-    def mem_usage_add(self, proxy: ProxyObject):
+    def mem_usage_add(self, proxy: ProxyObject) -> None:
         proxy_id = id(proxy)
         assert proxy_id not in self.proxy_id_to_dev_mems
         self.proxy_id_to_dev_mems[proxy_id] = set()
-        for dev_mem in proxy._pxy_get_device_memory_objects():
+        for dev_mem in get_device_memory_ids(proxy._pxy_get().obj):
             self.proxy_id_to_dev_mems[proxy_id].add(dev_mem)
             ps = self.dev_mem_to_proxy_ids[dev_mem]
             if len(ps) == 0:
-                self._mem_usage += sizeof(dev_mem)
+                self._mem_usage += dev_mem.nbytes
             ps.add(proxy_id)
 
-    def mem_usage_remove(self, proxy: ProxyObject):
+    def mem_usage_remove(self, proxy: ProxyObject) -> None:
         proxy_id = id(proxy)
         for dev_mem in self.proxy_id_to_dev_mems.pop(proxy_id):
             self.dev_mem_to_proxy_ids[dev_mem].remove(proxy_id)
             if len(self.dev_mem_to_proxy_ids[dev_mem]) == 0:
                 del self.dev_mem_to_proxy_ids[dev_mem]
-                self._mem_usage -= sizeof(dev_mem)
+                self._mem_usage -= dev_mem.nbytes
+
+    def buffer_info(self) -> List[Tuple[float, int, List[ProxyObject]]]:
+        ret = []
+        for dev_mem, proxy_ids in self.dev_mem_to_proxy_ids.items():
+            proxies = self.get_proxies_by_ids(proxy_ids)
+            last_access = max(p._pxy_get().last_access for p in proxies)
+            ret.append((last_access, dev_mem.nbytes, proxies))
```

Comment:
Unlike in the previous `buffer_info`, the list here seems to be made of multiple objects. I'm guessing you simply want to make the previous definition compatible?

Reply:
Yes, this is to standardize the interface, now we have `buffer_info` instead of `get_dev_access_info` and `get_host_access_info`